src/main/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessor.java

```diff
@@ -100,7 +100,11 @@ public void execute(IngestDocument ingestDocument, BiConsumer<IngestDocument, Ex
         try {
             validateEmbeddingFieldsValue(ingestDocument);
             Map<String, Object> knnMap = buildMapWithKnnKeyAndOriginalValue(ingestDocument);
-            mlCommonsClientAccessor.inferenceSentences(this.modelId, createInferenceList(knnMap), ActionListener.wrap(vectors -> {
+            List<String> inferenceList = createInferenceList(knnMap);
+            if (inferenceList.size() == 0) {
+                throw new IllegalArgumentException("Unable to process embedding since no text found from corresponding source fields");
+            }
```

Comment:
The inferenceList can be empty if the field which needs to inference is not present in the document. Please handle that use case in this PR.

Case when inference fields not present:
1. If for the document the field is genuinely not present.
2. In the update document use case the field to inference is not present as it doesn't need to be updated.

Reply:
@jmazanec15 @navneet1v Yeah, This is something I want to discuss with you guys. The error information in fact is not a real issue. Since the cases Navneet mentioned above, we should succeed the ingestion without any exception raised. This way user will never see the exception so no need to handle the proper prompt. I've changed the code in this PR, please take a look.

Reply:
@zane-neo 
I checked the code and it handles the case when user is updating the document, or even in the case when he is indexing the document for first time and he misses the fields. While this looks good. But the current code doesn't handle the case when user wants the ingestion request to fail because of lack of fields. I think we should think of that case too.

Reply:
@navneet1v These two are conflict cases, if we prompt the user lacking of fields which means we don't allow user's insert/update(particular without embedding fields). If we allow user to ingest without embedding fields, we shouldn't raise error to user. I prefer allowing user's ingestion without embedding fields, since this is a valid case for user(Some documents do not have embedding fields).

Reply:
@zane-neo The idea which I was thinking is let customer configure what they want to do in case the embedding fields are not present. One way to do this is by adding a parameter in the processor. But as of now I think we can go ahead with this change.